### DISCUSSION TITLE
The documentation is not clear as how to add it to existing project

### DISCUSSION THREAD

**Discussion by ddublon** [TanStack/router#3088] [Q&A]
i am having an existing project with firebase and reacttypescript but it is really not clear what to do to have the routing set up 

after installing the npm pachegase as in the docs and adding __root.tsx etc.. files then it is not clear what to do and how it works 

because in react router you would wrap the routes around the place where you want to be replaceable via the change in the route but here it is not clear how the flow works

**Comment by dolev146** [comment]
so if you try to migrate from React Router its pretty hard to understand how to
